### DISCUSSION TITLE
Union

### DISCUSSION THREAD

**Discussion by jgomez720** [KittyCAD/modeling-app#509] [New Feature Requests]
### Description
A tool that enables the user to union two separate bodies into one. This is also called "Combine" or "Unite" in other CAD packages. 

### Functionality
1. The user selects the _Union_ button. 
2. The user selects the two + bodies to unite (this should fail if the bodies are not intersecting).
3. Confirm.

### Supporting Docs/Images
https://github.com/KittyCAD/modeling-app/assets/114548659/9ff2199e-90dd-4eaa-99bf-04cb9b647108

## PseudoKCL
```
const part001 = startSketchOn('XY')
  |> startProfileAt([-12, 12], %)
  |> line([24, 0], %)
  |> line([0, -24], %)
  |> line([-24, 0], %)
  |> close(%)
  |> extrude(10, %)

const part002 = startSketchOn('XY')
  |> startProfileAt([6, 6], %)
  |> line([10, 0], %)
  |> line([0, -12], %)
  |> line([-10, 0], %)
  |> close(%)
  |> extrude(12, %)

const part003 = union({
  tools: [part001, part002]
})
```

**Expected Result:**
![Screenshot 2023-11-09 at 3 23 58 PM](https://github.com/KittyCAD/modeling-app/assets/114548659/12574885-700a-4144-a569-42411340aa25)
![Screenshot 2023-11-09 at 3 24 12 PM](https://github.com/KittyCAD/modeling-app/assets/114548659/b5b007b1-09b8-43db-a26b-211ebb61da78)

**Comment by yeroca** [comment]
I'm curious how you might define "intersecting" if the two solids are on opposite sides of the same plane.   For example, a unit cube whose bottom face is on the XY plane at [0,0,0], and a unit cube whose top face is on the XY plane at the same location.  The volume of the intersection is zero, but union() still makes logical sense in this case, at least to me.
